### PR TITLE
The integration of SasView

### PR DESCRIPTION
**Pull request by AtomAnu**
#### Description of Work
This is the integration of SasView. There are five main additions for this integration.

1. An additional example script, `example_runScripts_SasView.py` has been added to use               SasView as the software to fit the user-selected problem(s). At the start of the example script,   there are three condition statements with each statement verifying if one of the three required SasView packages is installed. Bumps, sasmodels and sascalc are the three required packages. As explained in the FitBenchmarking README.md file, Bumps and sasmodels can be installed using `pip` while sascalc is not an independent package.

   Additionally, the package sasmodels contains a script `data.py` which contains a module   called `load_data`. This module requires sascalc in order to function. In this module's code, the line of code which imports sascalc is written below.

   `from sas.sascalc.dataloader.loader import Loader`

   Furthermore, sascalc now exists under the directory `src/sas` which resides in the SasView repository (https://github.com/SasView/sasview)

   As a consequence, it was neccessary to copy the folder `sas` from the SasView repository and place it inside FitBenchmarking (`fitbenchmarking/sas`) in order to run SasView headless inside FitBenchmarking.

2. A parsing script, `parse_sasview_data.py` has been added to parse SasView problem definition files and their corresponding data files. This script can later be integrated with `parse_fitbenchmark_data.py`.

   in `parse_sasview_data.py`, the eval_f and get_function modules work in the same manner as that of `parse_nist_data.py` and `parse_fitbenchmark_data.py`. However, the module eval_f for `parse_sasview_data.py` accepts the parameters list in two formats. The first format is a string which contains the parameter names and values. The second format is to accept the parameter values as multiple parameters to the module.

   In `parse_fitbenchmark_data.py`, a new module, `get_bumps_function`, is created to format Mantid functions into the functions that are acceptable by Bumps. Functions/models that are acceptable by Bumps should not contain any arguments (*arg) or keyword arguments (**kwargs).

3. A data preparation script, `sasview/prepare_data.py`, has been created to prepare the data in the correct format. There are three main tasks done in this script. The first task is to set the data errors (data_e) if they are not given in the data file. The second task is to set the X data range if specified. If the X data range is not specified, the range would be set from the minimum X value to the maximum X value. The final task is to create a data object of type `sasmodels.data.Data1D` so that it can be passed onto Bumps.

4. A function definition preparation script, `sasview/func_def.py`, has been created to prepare the models/functions and their starting parameter values for Bumps fitting. The script also prepares the initial and the final function definition strings for each fit.

5. Finally, a script, `sasview/main.py`, has been added to perform fitting using Bumps. All the fitting occur in the module `fit`. One can observe that this module is quite lengthy. This is because the method of configuring the function/model is different for SasView models and other functions from Scipy and Mantid. Nevertheless, this module could be shorten up in the future.

   This script also calculates the chi-squared value and the run time of each fit. Then, a result object is created for the graph and table creating modules. A single result object contains the fitted Y values, the fit status, the chi-squared value, the fit runtime, the minimiser name, the initial function definition string and the final function definition string.

***
Note that the unit tests for these scripts have not been completed. 
#### Testing Instructions

1. run `example_runScripts.py` with problems of type NIST, FitBenchmark and SasView
2.  run `example_runScripts_mantid.py` with problems of type NIST, FitBenchmark and SasView
3. run `example_runScripts_SasView.py` with problems of type NIST, FitBenchmark and SasView

### CODE REVIEW

**Comment by Anders-Markvardsen** [comment]
@AtomAnu thanks for this PR and it detailed description. You write "As a consequence, it was neccessary to copy the folder sas from the SasView repository and place it inside FitBenchmarking (fitbenchmarking/sas) in order to run SasView headless inside FitBenchmarking.". 

A questions:

* In the sasview meeting tuesday did you discuss with the sasview team this aspects? - i.e. to include a copy of parts of there code until the sasview team has separated sascal out as separate package.
  * assuming this is OK then as a minimum please add readme in that folder to specify when the copy was made and a copy of your above reasoning for adding it.

Obviously the above is not ideal, but with the sasview team actively working on separating out sascal it seems like the best workaround

**Comment by Anders-Markvardsen** [comment]
Thinking about it again. As an alternative to copy and paste have you considered code which

* clone the sasview repo in read only mode, either a particular version of it directly or where after the repo has been cloned it is reset to a particular version. The latter is to guard against the expected future changes of this part of the sasview codebase 
* copy the sascal part, and this might be a tricky part: to a location where it is visible from the python path as appropriate 
* delete the cloned sasview repo

See https://stackoverflow.com/questions/2472552/python-way-to-clone-a-git-repository for suggestions for how a github repo can be cloned

File: README.md [hunk 1]
```diff
@@ -28,10 +28,10 @@ For help of how to use the command line/terminal, click the hyperlink correspond
 
 The above step is done to ensure that the compatible version of docutils packages (version 0.12) is installed. 
 
-7. Finally, in this terminal, run `example_scripts/example_runScript.py`, located in the fitbenchmarking folder. This example script fit benchmarks Mantid using all the available minimizers. The resulting tables can be found in `example_scripts/results`.
+7. Finally, in this terminal, run `example_scripts/example_runScript_mantid.py`, located in the fitbenchmarking folder. This example script fit benchmarks Mantid using all the available minimizers. The resulting tables can be found in `example_scripts/results`.
 
 ## FitBenchmarking Scipy
-The `example_runScripts.py` file can be changed such that it benchmarks minimizers supported by scipy instead of mantid (details provided in the file itself).
+The `example_runScripts.py` file is designed to benchmark minimizers supported by scipy instead of mantid (details provided in the file itself).
```
Comment (Anders-Markvardsen) [inline README.md]:
```suggestion
The `example_runScripts.py` file is designed to benchmark minimizers supported by software/libraries that provide straightforward cross-platform Python install; as of know this mean SciPy (more details provided in the file itself).
```

File: example_scripts/example_runScripts.py [hunk 2]
```diff
@@ -77,7 +78,7 @@
 # problem_sets = ["Neutron_data", "NIST/average_difficulty"]
 # problem_sets = ["CUTEst", "Muon_data", "Neutron_data", "NIST/average_difficulty", "NIST/high_difficulty", "NIST/low_difficulty"]
 
-problem_sets = ["NIST/low_difficulty"]
+problem_sets = ["Muon_data"]
```
Comment (Anders-Markvardsen) [inline example_scripts/example_runScripts.py]: @AtomAnu did you mean to change problem set up Muon_data in this script? I.e. with problem_sets = Muon_data this script with not run without mantid

**Comment by AtomAnu** [comment]
Dear Anders,

The intention was for the purpose of testing. The problem_sets variable
should be set to [“NIST/low_difficulty”].


Kind regards,
Atom

**Comment by Anders-Markvardsen** [comment]
Note I observe
```
C:\Midlertiddig\fitbenchmarking\example_scripts>python example_runScripts_SasView.py

Running the benchmarking on the 1D_data problem set

Traceback (most recent call last):
  File "example_runScripts_SasView.py", line 117, in <module>
    use_errors=use_errors, results_dir=results_dir)
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\fitting_benchmarking.py", line 65, in do_fitting_benchmark
    prob_results = do_benchmarking(user_input, problem_groups, group_name)
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\fitting_benchmarking.py", line 85, in do_benchmarking
    prob_results.append(do_fitbm_group(user_input, block))
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\fitting_benchmarking.py", line 115, in do_fitbm_group
    problem = parse.parse_problem_file(prob_file)
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\parsing\parse.py", line 59, in parse_problem_file
    problem = parse_sasview_data.FittingProblem(prob_file)
  File "C:\Midlertiddig\fitbenchmarking\fitbenchmarking\parsing\parse_sasview_data.py", line 33, in __init__
    data_obj = load_data(data_file_path)
  File "C:\MantidInstall\bin\lib\site-packages\sasmodels\data.py", line 54, in load_data
    from sas.sascalc.dataloader.loader import Loader  # type: ignore
  File "C:\Midlertiddig\fitbenchmarking\sas\sascalc\dataloader\loader.py", line 36, in <module>
    from .readers import cansas_reader
  File "C:\Midlertiddig\fitbenchmarking\sas\sascalc\dataloader\readers\cansas_reader.py", line 13, in <module>
    from lxml import etree
ImportError: No module named lxml
```

**Comment by AtomAnu** [comment]
@Anders-Markvardsen I forgot to include that you would be required to install the package **lxml** (`pip install lxml`) as it is required by sascalc in the case when it is reading .xml files.

**Comment by Anders-Markvardsen** [comment]
Concerning the new folder in benchmark_problems '1D_data' I would suggest changing the structure to

SAS_modelling/1D/data_files/
                               prob_def_1.txt
                               prob_def_2.txt

SAS stands for Small Angle Scattering and the examples/problems in that folder are specifically for modelling SAS data

**Comment by AtomAnu** [comment]
@Anders-Markvardsen 
I have just added the following changes:
1. Updated the FitBenchmarking README.md file in the section, **FitBenchmarking SasVIew**, to include the instruction to install **lxml**.
2. Added a README.md file inside the folder **sas** to explain why `sascalc` is required in FItBenchmarking and why it is added here temporarily.
3. Edited the format of log messages sent when fits fail. The level of the log messages has also been changed to `warning` instead of `error`.
4. Changed the name of folder containing 1D SasView problem files from `1D_data` to `1D`. The location of the folder is also changed. It is now located in a folder called `SAS_modelling`.
5. Changed the folder names, `Muon_data` and `Neutron_data`, to `Muon` and `Neutron`, respectively.

File: example_scripts/SasView_example.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+from sasmodels.core import load_model
```
Comment (Anders-Markvardsen) [inline example_scripts/SasView_example.py]: What is the purpose of this file and how does it differ from example_runScripts_SasView.py

File: fitbenchmarking/fitbenchmark_one_problem.py [hunk 4]
```diff
@@ -58,8 +58,11 @@ def fitbm_one_prob(user_input, problem):
     for function in function_definitions:
         # Ad hoc exception for running the scipy script
         # scipy does not currently support the GEM problem
-        if 'GEM' in problem.name and user_input.software == 'scipy':
-            break
+        # if 'GEM' in problem.name and user_input.software == 'scipy':
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitbenchmark_one_problem.py]: please remove dead code comments

**Comment by AtomAnu** [comment]
That script is a script in which I have added in order to test running
SasView headless from the start to the end. I have forgotten to remove it
when creating the pull request.

File: fitbenchmarking/fitting/mantid/externals.py [hunk 5]
```diff
@@ -61,9 +61,15 @@ def set_ties(function_object, ties):
 
     @returns :: mantid function object with ties
     """
+
     for idx, ties_per_func in enumerate(ties):
         for tie in ties_per_func:
-            exec "function_object.tie({'f" + str(idx) + "." + tie + "})"
+            exec("param_dict = {'f" + str(idx) + "." + tie + "}")
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/mantid/externals.py]: param_dict appears to be declared but not used. Please delete commented out lines of code that are no longer needed. Please add a code comment to explain this more complex code

File: fitbenchmarking/fitting/mantid/func_def.py [hunk 6]
```diff
@@ -100,6 +102,47 @@ def function1D(self, xdata):
 
     return function_defs
 
+def parse_sasview_function_definitions(problem):
+    """
+    Helper function that parses the SasView function definitions and
+    transforms them into a mantid-readable format.
+
+    @param problem :: object holding the problem information
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/mantid/func_def.py]: change to 'object holding the problem information in SasView format' ?

File: fitbenchmarking/fitting/mantid/func_def.py [hunk 7]
```diff
@@ -100,6 +102,47 @@ def function1D(self, xdata):
...
+
+    @returns :: the formatted function definition (str)
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/mantid/func_def.py]: change to 'Mantid formatted function definition (str)'  ?

File: fitbenchmarking/fitting/mantid/main.py [hunk 8]
```diff
@@ -152,7 +152,7 @@ def parse_result(fit_result, t_start, t_end):
         fit_wks = fit_result.OutputWorkspace
         fin_function_def = str(fit_result.Function)
         runtime = t_end - t_start
-
+    print(fin_function_def)
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/mantid/main.py]: Do you mean to have this print statement here?

File: fitbenchmarking/fitting/plotting/plots.py [hunk 9]
```diff
@@ -230,3 +233,29 @@ def get_mantid_starting_guess_data(wks_created, function, problem):
...
+
+def get_sasview_starting_guess_data(data_struct, problem, function):
+    """
+
+    :param data_struct:
+    :param function:
+    :return:
+    """
+
+    # yData = function[0](data_struct.x)
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/plotting/plots.py]: dead comment?

File: fitbenchmarking/fitting/plotting/plots.py [hunk 10]
```diff
@@ -230,3 +233,29 @@ def get_mantid_starting_guess_data(wks_created, function, problem):
...
+
+    yData = problem.eval_f(data_struct.x, function[1])
+    # from sasmodels.bumps_model import Experiment, Model
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/plotting/plots.py]: dead comment lines?

**Comment by AtomAnu** [comment]
That module is currently in progress of investigation regarding the issue
of setting ties on mantid. It was initially used for the module ties for
mantid. However, the modules ties does not seem to set the specified ties.
Thus, I have tried to use the module fix instead as all the current ties
specified in the problem definition files are for the purpose of fixing
some parameter values.

I have paused the investigation at the moment as the module fix does not
work as well when fitting mantid problems with Scipy and SasView.

Could you give me an advice of what I should do? Should I remove any
statements that call this module?

File: fitbenchmarking/fitting/plotting/plots.py [hunk 11]
```diff
@@ -230,3 +233,29 @@ def get_mantid_starting_guess_data(wks_created, function, problem):
     yData = tmp.readY(1)
 
     return xData, yData
+
+def get_sasview_starting_guess_data(data_struct, problem, function):
+    """
+
+    :param data_struct:
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/plotting/plots.py]: please add some minimum info about input parameters and the return output

File: fitbenchmarking/fitting/prerequisites.py [hunk 12]
```diff
@@ -45,6 +45,8 @@ def prepare_software_prerequisites(software, problem, use_errors):
         return prepare_mantid(problem, use_errors)
     elif software == 'scipy':
         return prepare_scipy(problem, use_errors)
+    elif software == 'sasview':
+        return prepare_sasview(problem, use_errors)
     # elif software == 'your_software':
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/prerequisites.py]: dead comments?

File: fitbenchmarking/fitting/prerequisites.py [hunk 13]
```diff
@@ -95,3 +97,14 @@ def prepare_scipy(problem, use_errors):
         problem.start_x = - np.inf
         problem.end_x = np.inf
     return data, cost_function, function_definitions
+
+def prepare_sasview(problem, use_errors):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/prerequisites.py]: prepare sasview for what? Please explain in comment

**Comment by AtomAnu** [comment]
Sure! I’ll change the parts which you have mentioned accordingly.

File: fitbenchmarking/fitting/sasview/func_def.py [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+
+from utils.logging_setup import logger
+from sasmodels.core import load_model
+import numpy as np
+import re
+
+
+def function_definitions(problem):
+    """
+    Transforms the prob.equation field into a function that can be
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/func_def.py]: Is this def description correct - i.e. does it match what this def does?

File: fitbenchmarking/fitting/sasview/func_def.py [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+    @param problem :: object holding the problem information
+
+    @returns :: the type of the problem in capital letters (e.g. NIST)
+    """
+    problem_file_name = problem.__class__.__module__
+    problem_type = (problem_file_name.split('_')[1]).upper()
+
+    return problem_type
+
+def get_fin_function_def(final_param_values, problem, init_func_def):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/func_def.py]: what does this def do? and what does 'fin' in get_fin_function_def stand for?

File: fitbenchmarking/fitting/sasview/func_def.py [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+        str(round(next(rep), 3)), init_func_def)
+        if len(all_attributes) != 0:
+            fin_function_def = [fin_function_def.replace('+', attr) for attr in all_attributes]
+
+    return fin_function_def
+
+
+def get_init_function_def(function, problem):
+    """
+    Get the initial function definition string.
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/func_def.py]: Do you mean 'Create and return the initial function definition string suitable to be passed to bumps.' ?

File: fitbenchmarking/fitting/sasview/func_def.py [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+
+    @param function :: array containing the function information
+    @param problem :: object holding the problem information
+
+    @returns :: the initial function definition string
+    """
+
+    problem_type = extract_problem_type(problem)
+
+    if not 'name=' in str(problem.equation):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/func_def.py]: Please add code comment to explain which case 'if not 'name=' in str(problem.equation):' refers to

File: fitbenchmarking/fitting/sasview/func_def.py [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+        params = function[0].__code__.co_varnames[1:]
+        param_string = ''
+        for idx in range(len(function[1])):
+            param_string += params[idx] + "= " + str(function[1][idx]) + ", "
+        param_string = param_string[:-2]
+        init_function_def = function[2] + " | " + param_string
+    elif problem_type == 'SasView'.upper():
+        init_function_def = problem.equation + ',' + problem.starting_values
+        init_function_def = re.sub(r"(=)([-+]?\d+)([^.\d])", r"\g<1>\g<2>.0\g<3>", init_function_def)
+    else:
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/func_def.py]: And which case this refers to

File: fitbenchmarking/fitting/sasview/func_def.py [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+    if not 'name=' in str(problem.equation):
+        params = function[0].__code__.co_varnames[1:]
+        param_string = ''
+        for idx in range(len(function[1])):
+            param_string += params[idx] + "= " + str(function[1][idx]) + ", "
+        param_string = param_string[:-2]
+        init_function_def = function[2] + " | " + param_string
+    elif problem_type == 'SasView'.upper():
+        init_function_def = problem.equation + ',' + problem.starting_values
+        init_function_def = re.sub(r"(=)([-+]?\d+)([^.\d])", r"\g<1>\g<2>.0\g<3>", init_function_def)
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/func_def.py]: this looks tricky - a code comment would help

File: fitbenchmarking/fitting/sasview/main.py [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+        individual_result = \
+                misc.create_result_entry(problem, status, chi_sq, runtime, minimizer,
+                                     init_func_def, fin_func_def)
+
+        results_problem.append(individual_result)
+
+    return results_problem, best_fit
+
+def fit(problem, data, function, minimizer, init_func_def):
+    """
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/main.py]: code comment please

Comment (Anders-Markvardsen) [reply fitbenchmarking/fitting/sasview/main.py]: I meant def comment please

File: fitbenchmarking/fitting/sasview/main.py [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+
+    @returns ::  the status, either success or failure (str), the data
+                of the fit, the final function definition and the
+                runtime of the fitting software
+    """
+
+    t_start, t_end = None, None
+    model = function[0]
+
+    if hasattr(model, '__call__'):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/main.py]: A few code comments here may help

File: fitbenchmarking/fitting/sasview/main.py [hunk 22]
```diff
@@ -0,0 +1,171 @@
...
+    except (RuntimeError, ValueError) as err:
+        logger.warning("Fit failed: " + str(err))
+
+    status = 'success' if result.success is True else 'failed'
+
+    fitted_y = func_wrapper.theory()
+
+    final_param_values = result.x
+
+    fin_func_def = get_fin_function_def(final_param_values, problem, init_func_def)
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/main.py]: what does 'fin' in fin_func_def stand for

File: fitbenchmarking/fitting/sasview/prepare_data.py [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+# <https://github.com/mantidproject/fitbenchmarking>.
+# Code Documentation is available at: <http://doxygen.mantidproject.org>
+from __future__ import (absolute_import, division, print_function)
+
+import numpy as np
+from sasmodels.data import empty_data1D, Data1D
+
+from utils.logging_setup import logger
+
+def prepare_data(problem, use_errors):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/sasview/prepare_data.py]: prepare data for what?

File: fitbenchmarking/fitting/sasview/main.py (hunk 22 above)
Comment (AtomAnu) [reply fitbenchmarking/fitting/sasview/main.py]: The word fin stands for final in this case.

File: fitbenchmarking/fitting/scipy/func_def.py [hunk 24]
```diff
@@ -51,37 +51,50 @@ def get_fin_function_def(init_function_def, func_callable, popt):
 
     @returns :: the final function definition string
     """
-    if not 'name=' in str(func_callable):
+    if not 'name=' in init_function_def:
         popt = list(popt)
         params = init_function_def.split("|")[1]
-        params = re.sub(r"[-+]?\d+\.\d+", lambda m, rep=iter(popt):
+        params = re.sub(r"[-+]?\d+[.]\d+", lambda m, rep=iter(popt):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/scipy/func_def.py]: looks subtle - code comment please and e.g. line 61 also

File: fitbenchmarking/fitting/scipy/func_def.py [hunk 25]
```diff
@@ -51,37 +51,50 @@ def get_fin_function_def(init_function_def, func_callable, popt):
...
+    @param equation :: the string containing the function
+                                definition in mantid/sasview format
 
     @returns :: the initial function definition string
     """
-    if not 'name=' in str(function[0]):
+
+    problem_type = extract_problem_type(problem)
+
+    if not 'name=' in str(problem.equation):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/fitting/scipy/func_def.py]: please add comment to specify what use case this is. And for else below also

File: fitbenchmarking/parsing/fitbenchmark_data_functions.py [hunk 26]
```diff
@@ -50,11 +50,36 @@ def fitbenchmark_func_definitions(functions_string):
         fit_function = make_fitbenchmark_fit_function(name, fit_function, params_set)
     fit_function = set_ties(fit_function, ties)
 
-
     function_defs = [[fit_function, params]]
 
     return function_defs
 
+def get_fit_function_without_kwargs(fit_function, functions_string):
+    """
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/parsing/fitbenchmark_data_functions.py]: please add short def description

File: fitbenchmarking/parsing/fitbenchmark_data_functions.py [hunk 27]
```diff
@@ -50,11 +50,36 @@ def fitbenchmark_func_definitions(functions_string):
...
+
+    :param fit_function:
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/parsing/fitbenchmark_data_functions.py]: some parameter info please

File: fitbenchmarking/parsing/fitbenchmark_data_functions.py [hunk 28]
```diff
@@ -179,6 +204,7 @@ def get_fitbenchmark_ties(param_set, ties):
         else:
             tie = param_set[start + 1:comma]
         ties_per_function.append(tie.replace("=", "': "))
+        # ties_per_function.append(tie)
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/parsing/fitbenchmark_data_functions.py]: dead comment?

File: fitbenchmarking/parsing/parse_fitbenchmark_data.py [hunk 29]
```diff
@@ -59,7 +59,7 @@ def __init__(self, fname):
         #String containing the function name(s) and the starting parameter values for each function
         self._equation = entries['function']
 
-        self._starting_values = None
+        # self._starting_values = (entries['function'].split(',', 1))[1]
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/parsing/parse_fitbenchmark_data.py]: dead comment?

**Comment by Anders-Markvardsen** [comment]
All definitions works with Scipy and Mantid. All work for SasView except for FitBenchmark type problems. For example by setting problem_sets = ["Neutron"] in example_runscript_sasview gives the error

```
C:\Midlertiddig\fitbenchmarking\fitbenchmarking\fitting\sasview\main.py in fit(problem, data, function, minimizer, init_func_def)
    107             # The problem type is NIST
    108             #Formatting the function parameter names
--> 109             formatted_param_names = [param[0] for param in problem.starting_values]
    110
    111         param_values = function[1]

TypeError: 'NoneType' object is not iterable
```

**Comment by Anders-Markvardsen** [comment]
@AtomAnu thanks for the changes and again for your work over the summer.

I have created work for follow up work that this PR has highlighted during testing.

**Pull request merged**